Incremental search

zumi-n opened the pull request:
#WHAT
インクリメンタルサーチ実装。
https://gyazo.com/6dc7cc8d463e8cc586d4b219609da61e

#WHY
インクリメンタルサーチ実装のため。

zumi-n committed:
インクリメンタルサーチ実装

ユーザー検索

zumi-n committed:
インクリメンタルサーチ

kohei-isoda commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,51 @@
+      var html = buildHTML(message);
+      $('.messages').append(html)
+      $('.form__message').val('')
+      $('.form__submit').prop('disabled', false);
```
failにも書かれているので[DRYの法則](https://xn--97-273ae6a4irb6e2hsoiozc2g4b8082p.com/%E3%82%A8%E3%83%83%E3%82%BB%E3%82%A4/DRY%E5%8E%9F%E5%89%87/)に従ってalwaysなどの関数を使って処理を纏めましょう！

zumi-n replied:
$('.form__submit').prop('disabled', false);を.always()でまとめました。

kohei-isoda commented on app/assets/javascripts/user.js:
```diff
@@ -0,0 +1,76 @@
+
+    $(document).on('click', '.user-search-add', function(){
+      $(this).parent()[0].remove();
+      var user_id = $(this).attr("data-user-id");
```
attrではなくdataメソッドを使いましょう！カスタムデータ属性を使用している利点が薄れてしまうので！

zumi-n replied:
var user_id = $(this).attr("data-user-id”);
var user_id = $(this).attr("data-user-name”);をdataメソッドに変更
var user_id = $(this).data("user-id");
var user_name = $(this).data("user-name");

kohei-isoda commented on app/controllers/users_controller.rb:
```diff
@@ -1,7 +1,13 @@
 class UsersController < ApplicationController
-  def
+  def index
+    @users = User.where('name LIKE(?)', "%#{params[:keyword]}%")
```
こちらにもcurrent_userが入らないようにしたいです！

zumi-n replied:
@users = User.where('name LIKE(?)', "%#{params[:keyword]}%")を以下に変更

@users = User.where('name LIKE(?) and id != ?', "%#{params[:keyword]}%", current_user)

kohei-isoda commented on app/views/messages/create.json.jbuilder:
```diff
@@ -0,0 +1,5 @@
+json.name    @message.user.name
+json.content    @message.content
+json.time    @message.created_at
```
日付のフォーマット指定をきちんとしましょう！
rails time formatのように調べて頂けると出てきます！

zumi-n replied:
json.time    @message.created_atを以下に変更
json.time    @message.created_at.strftime('%Y/%m/%d/ %H:%M')

kohei-isoda reviewed (commented):
ご提出有難うございます、そして、機能実装お疲れ様です！
ただ、カリキュラムのメッセージ非同期通信とインクリメンタルサーチの実装が此のブランチに入ってしまっているので本来であれば2つに分ける必要があります。1機能につき1Pull Requestが守れるようにしましょう！

zumi-n committed:
Edit インクリメンタルサーチ

#WHAT
・app/assets/javascripts/message.js
$('.form__submit').prop('disabled', false);を.always()でまとめる

・app/assets/javascripts/user.js
var user_id = $(this).attr("data-user-id”);をdataメソッドに変更

・ app/controllers/users_controller.rb
@users = User.where('name LIKE(?)', "%#{params[:keyword]}%")
current_user入れないように修正

・app/views/messages/create.json.jbuilder
json.time    @message.created_at 日付フォーマット修正

#WHY
レビュー反映のため

zumi-n committed:
Edit message.js

#WHAT
.fail、.alwaysの末尾の;とる

#WHY
スクロール機能反応しなかったため

zumi-n commented:
#WHAT
・app/assets/javascripts/message.js
$('.form__submit').prop('disabled', false);を.always()でまとめる

・app/assets/javascripts/user.js
var user_id = $(this).attr("data-user-id”);をdataメソッドに変更

・ app/controllers/users_controller.rb
@users = User.where('name LIKE(?)', "%#{params[:keyword]}%")
current_user入れないように修正

・app/views/messages/create.json.jbuilder
json.time    @message.created_at 日付フォーマット修正

https://gyazo.com/4986f088deb49d01e9967a93601de216
https://gyazo.com/1133f888e92b5cc8e68c22f31633acb2

#WHY
レビュー反映のため

miyagit commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,54 @@
+    .done(function(message){
+      var html = buildHTML(message);
+      $('.messages').append(html)
+      $('.form__message').val('')
```
こちらform属性のresetメソッドを使用しましょう。

zumi-n replied:
resetメソッド使用に変更

miyagit commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,54 @@
+   })
+      .fail(function(){
+      alert('メッセージを入力してください');
+      // $('.form__submit').prop('disabled', false);
```
こちらコメントアウトしている理由はありますでしょうか？

zumi-n replied:
消し忘れのため削除しました。

kuroko918 commented:
修正が反映されていません。
確認お願いします！

zumi-n committed:
[WIP]edit message.js

zumi-n commented:
#WHAT
指摘箇所取り込み

#WHY
修正するため

kuroko918 commented on app/views/groups/_form.html.haml:
```diff
@@ -7,35 +7,27 @@
-      <div id='user-search-result'></div>
-      </div>
+    .chat-group-form__field--left
+      = f.label :users, "チャットメンバーを追加", class: 'chat-group-form__label', for:'user-search-field'
```
このlabelに対応するinputがないですが、labelタグである必要はありますか？

kuroko918 commented on app/views/groups/_form.html.haml:
```diff
@@ -7,35 +7,27 @@
   .chat-group-form__field.clearfix
     .chat-group-form__field--left
-      %label.chat-group-form__label{:for => "chat_group_チャットメンバー"} チャットメンバー
+      %label.chat-group-form__label{:for => "members"} チャットメンバー
```
このlabelに対応するinputがないですが、labelタグである必要はありますか？

kuroko918 commented on config/routes.rb:
```diff
@@ -1,7 +1,7 @@
+  root 'messages#index'
+  resources :users, only: [:index, :edit, :update]
   resources :groups, only: [:new, :create, :edit, :update] do
     resources :messages, only: [:index, :create]
```
rootにmessages#indexを指定しているので、onlyのindexは消しましょう！

zumi-n committed:
Edit インクリメンタルサーチ

zumi-n commented:
#WHAT
指摘箇所取り込み

#WHY
修正するため

kuroko918 commented on app/controllers/messages_controller.rb:
```diff
@@ -28,7 +28,7 @@ def message_params
 
   def set_group
-    @group = Group.find(params[:group_id])
+    @group = Group.find_by(params[:group_id])
```
これで動きますか？findでないといけないかと思いますが。

zumi-n replied:
findにしても動くように修正しました。

zumi-n committed:
Messageコントローラー修正

@group = Group.find_by(params[:group_id])
↓
@group = Group.find(params[:group_id])でも動くように修正

kuroko918 commented:
LGTMです！
